### PR TITLE
Fe/chat manage UI

### PR DESCRIPTION
**Pull request by pluz-stas**

### CODE REVIEW

File: SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                Name = chatName,
+                Photo = photo,
+            };
+
+            await Http.PostAsync($"api/chats/", chat);
+            OnClose.InvokeAsync();
+        }
+
+        private long maxFileSize = 1024 * 1024;
+        async Task LoadFile(InputFileChangeEventArgs e)
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs]: Please add modifier

File: SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                IsPublic = isPublic,
+                Name = chatName,
+                Photo = photo,
+            };
+
+            await Http.PostAsync($"api/chats/", chat);
+            OnClose.InvokeAsync();
+        }
+
+        private long maxFileSize = 1024 * 1024;
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs]: Put all fields in top of file

File: SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        private string GetImg()
+        {
+            if (photo != null)
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs]: Use string.IsNullOrWhiteSpace

File: SimpleChat.Client/Components/Modals/Chat/EditChatModal.razor.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+        private async Task EditAsync()
+        {
+            await Http.PutAsync($"api/chats/{Chat.Id}", Chat);
+            OnClose.InvokeAsync();
+        }
+
+        private long maxFileSize = 1024 * 1024;
+        async Task LoadFile(InputFileChangeEventArgs e)
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/EditChatModal.razor.cs]: Add access modifier

File: SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            try
+            {
+                using var reader =
+                    new StreamReader(e.File.OpenReadStream(maxFileSize));
+                var ms = new MemoryStream();
+                await reader.BaseStream.CopyToAsync(ms);
+                photo = ms.ToArray();
+                StateHasChanged();
+            }
+            catch
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs]: Is it necessary?

File: SimpleChat.Client/Components/Modals/Chat/EditChatModal.razor [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                <div class="hidden-input">
+                    <InputFile OnChange="@LoadFile" accept="image/jpeg"/>
+                </div>
+            </div>
+
+            <div class="d-flex flex-column p-3">
+                <input class="input-name" @bind="Chat.Name" type="text">
+                <div class="d-flex flex-row">
+                    <input @bind="Chat.IsPublic" type="checkbox">
+                    <div>is public</div>
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/EditChatModal.razor]: Add localize

File: SimpleChat.Client/Pages/ChatPage.razor.cs [hunk 7]
```diff
@@ -51,5 +55,29 @@ private async Task SendAsync()
 
             messageInput = string.Empty;
         }
+        
+        string GetAvatarName(string str)
```
Comment (chelovek1418) [inline SimpleChat.Client/Pages/ChatPage.razor.cs]: modifier

File: SimpleChat.Client/Pages/ChatPage.razor.cs [hunk 8]
```diff
@@ -51,5 +55,29 @@ private async Task SendAsync()
...
+                StringSplitOptions.RemoveEmptyEntries).Select(s => s[0]).ToArray()).ToUpper();
+            if (acronym.Length > 2)
+            {
+                acronym = acronym.Substring(0, acronym.Length - (acronym.Length - 2));
+            }
+            
+            return acronym;
+        }
+
+        private string GetImg()
```
Comment (chelovek1418) [inline SimpleChat.Client/Pages/ChatPage.razor.cs]: I see same logic in different places. Put it in one

File: SimpleChat.Client/Services/HttpClientService.cs [hunk 9]
```diff
@@ -60,6 +60,25 @@ public async Task PostAsync<T>(string uri, T value)
             }
         }
 
+        public async Task PutAsync<T>(string uri, T value)
```
Comment (chelovek1418) [inline SimpleChat.Client/Services/HttpClientService.cs:66]: This method needs refactoring

File: SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor [hunk 10]
```diff
@@ -0,0 +1,38 @@
+﻿@inject IStringLocalizer<Resource> localizer
+
+<div class="modal-background" role="dialog" aria-modal="true">
+    <section class="chat-modal">
+        <div class="modal-head">
+            <h5>@localizer[Resource.Settings]</h5>
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor]: This is not settings

File: SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs (hunk 3 above)
Comment (pluz-stas) [reply SimpleChat.Client/Components/Modals/Chat/CreateChatModal.razor.cs]: photo variable is byte[]

File: SimpleChat.Client/Components/Modals/Chat/EditChatModal.razor.cs [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        public ChatContract Chat { get; set; }
+
+
+        private async Task EditAsync()
+        {
+            await Http.PutAsync($"api/chats/{Chat.Id}", Chat);
+            OnClose.InvokeAsync();
+        }
+
+        private async Task LoadFile(InputFileChangeEventArgs e)
```
Comment (chelovek1418) [inline SimpleChat.Client/Components/Modals/Chat/EditChatModal.razor.cs]: Put this logic into external service

**Pull request merged**